misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -4473,6 +4483,9 @@ private void preserveIndentation(boolean value) {
     private int getPreservedIndentation(Token token) {
         int position = token.lineRange().startLine().offset();
         int offset = position % 4;
+        if (env.currentIndentation % 4 == 0 && env.currentIndentation > position) {
```

Comment:
Here, Is this number a constant or configurable? I guess this is the default WS tab length. Better to move this to either option or to a constant. The same applies to the line above and bellow the if condition.